felix/bpf-gpl/jump.h

```diff
@@ -120,7 +120,19 @@ CALI_MAP_V1(cali_jump_prog_map, BPF_MAP_TYPE_PROG_ARRAY, __u32, __u32, 2400, 0)
 	bpf_tail_call((ctx)->xdp, &cali_jump_prog_map, (ctx)->xdp_globals->jumps[PROG_INDEX_POLICY])
 #else /* CALI_F_XDP */
 
-#define cali_jump_prog_map map_symbol(cali_jump, 3)
+#if CALI_F_HEP || CALI_F_PREAMBLE || CALI_F_DEF_POLICY
+#if CALI_F_INGRESS
```

Comment:
How about moving the nested conditional to bpf.h and using it to #define some new constants:

```
#if CALI_F_HEP || CALI_F_PREAMBLE || CALI_F_DEF_POLICY
// For HEPs policy direction (CALI_F_INGRESS) matches attachment direction.
#if CALI_F_INGRESS
#define CALI_HOOK_INGRESS
#else
#define CALI_HOOK_EGRESS
#endif
#else
// For WEPs, the policy direction is opposite to the tc hook that the
// program is attached to.
#if CALI_F_INGRESS
#define CALI_HOOK_EGRESS
#else
#define CALI_HOOK_INGRESS
#endif
#endif
```
Then here, you'd just need this, which might be less prone to misunderstanding:
```
#ifdef CALI_HOOK_INGRESS
#define cali_jump_prog_map map_symbol(cali_jump_ing, 2)
#else
#define cali_jump_prog_map map_symbol(cali_jump_egr, 2)
#endif
```